### PR TITLE
[FABCAG-22] Add compatibility for fabric and go versions

### PR DESCRIPTION
**Pull request by awjh-ibm**
[FABCAG-22](https://jira.hyperledger.org/browse/FABCAG-22)

Adds:
- Compatibility matrix

### CODE REVIEW

File: COMPATIBILITY.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- The version of Go used to build the code
+- When starting a chaincode container to run a Smart Contract the version of Go used to build the Smart Contract is determined by these factors:
+
+Fabric v2.0 will, by default, start up a docker image to host the chaincode and contracts. The version of Go used is therefore determined by Fabric.
+
+With Fabric v2.0.0, the chaincode container can be configured to be started by other means, and not the Peer. In this case, the version of Go used to build the container is not in the control of Fabric.
+
+Packages that are produced are produced are: `contractapi`, `metadata` & `serializer`
+
+## Supported Go versions
```
Comment (heatherlp) [inline COMPATIBILITY.md]: do you need this section?

File: COMPATIBILITY.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+The key elements are:
+- The version of fabric-contract-api-go packages used
+- The version of Go used to build the code
+- When starting a chaincode container to run a Smart Contract the version of Go used to build the Smart Contract is determined by these factors:
+
+Fabric v2.0 will, by default, start up a docker image to host the chaincode and contracts. The version of Go used is therefore determined by Fabric.
+
+With Fabric v2.0.0, the chaincode container can be configured to be started by other means, and not the Peer. In this case, the version of Go used to build the container is not in the control of Fabric.
+
+Packages that are produced are produced are: `contractapi`, `metadata` & `serializer`
```
Comment (heatherlp) [inline COMPATIBILITY.md]: you've got one too many producers

File: COMPATIBILITY.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+GitHub is used for code base management, issues should be reported in the [FABCAG](https://jira.hyperledger.org/projects/FABCAG/issues) component in JIRA.
+
+## Summary of Compatibility
+
+This table shows the summary of the compatibility of the package at version 2.0, together with the Go version it requires and the Fabric Peer version it can communicate with.
+
+|                       | Peer Connectivity v2.0 | Go 1.13 |
+| --------------------- | ---------------------- | ------- |
+| Packages **v2.0**     | Yes                    | Yes     |
```
Comment (heatherlp) [inline COMPATIBILITY.md]: I'm more in favour of having a "Go" column, with the entry in the table being the value that is supported, what do you think?
i.e. https://github.com/hyperledger/fabric-gateway-java#compatibility

File: COMPATIBILITY.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
```
Comment (heatherlp) [inline COMPATIBILITY.md]: Do we explicitly want to mention we're not supporting v1.4? I'm wondering whether it's worth getting Lesley-Ann's thoughts?

File: COMPATIBILITY.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+## Compatibility
+
+The key elements are:
+- The version of fabric-contract-api-go packages used
+- The version of Go used to build the code
+- When starting a chaincode container to run a Smart Contract the version of Go used to build the Smart Contract is determined by these factors:
+
+Fabric v2.0 will, by default, start up a docker image to host the chaincode and contracts. The version of Go used is therefore determined by Fabric.
+
+With Fabric v2.0.0, the chaincode container can be configured to be started by other means, and not the Peer. In this case, the version of Go used to build the container is not in the control of Fabric.
```
Comment (heatherlp) [inline COMPATIBILITY.md]: you flip between v2.0 and v2.0.0 of fabric, do you intend to?

File: COMPATIBILITY.md (hunk 2 above)
Comment (heatherlp) [reply COMPATIBILITY.md]: why is this line relevant for compatibility?

File: COMPATIBILITY.md (hunk 4 above)
Comment (awjh-ibm) [reply COMPATIBILITY.md]: It probably is worth asking her to cast an eye over it. I have now updated the table so it may make it clearer

**Pull request merged**